e2e/deployment.go

```diff
@@ -296,3 +297,169 @@ func (rnr *rookNFSResource) Do(action kubectlAction) error {
+	e2elog.Logf("waiting for deployment updates %s/%s", ns, deploymentName)
+
+	// Scale down to 0.
+	scale, err := c.AppsV1().Deployments(ns).GetScale(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get scale deployment %s/%s: %w", ns, deploymentName, err)
+	}
+	count := scale.Spec.Replicas
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 0
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Update deployment.
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get deployment %s/%s: %w", ns, deploymentName, err)
+	}
+	cid := deployment.Spec.Template.Spec.Containers //cid: read as containers in deployment
+	for i := range cid {
+		if contains(containers, cid[i].Name) {
+			match := false
+			for j, ak := range cid[i].Args {
+				if ak == key {
+					// do replacement of value
+					match = true
+					cid[i].Args[j] = fmt.Sprintf("--%s=%s", key, value)
+					break
+				}
+			}
+			if !match {
+				// append a new key value
+				cid[i].Args = append(cid[i].Args, fmt.Sprintf("--%s=%s", key, value))
+			}
+			deployment.Spec.Template.Spec.Containers[i].Args = cid[i].Args
+		}
+	}
+	// clear creationTimestamp, generation, resourceVersion, and uid
+	deployment.CreationTimestamp = metav1.Time{}
+	deployment.Generation = 0
+	deployment.ResourceVersion = "0"
+	deployment.UID = ""
+	err = waitForDeploymentUpdate(c, deployment, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Scale up to count.
+	scale.Spec.Replicas = count
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// wait for scale to become count
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err = wait.PollImmediate(poll, t, func() (bool, error) {
+		deploy, getErr := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if getErr != nil {
+			if isRetryableAPIError(getErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment Get %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error getting deployment %s/%s: %w", ns, deploymentName, getErr)
+		}
+		if deploy.Status.Replicas != count {
+			e2elog.Logf("Expected deployment %s/%s replicas %d, got %d", ns, deploymentName, count, deploy.Status.Replicas)
+
+			return false, fmt.Errorf("error expected deployment %s/%s replicas %d, got %d",
+				ns, deploymentName, count, deploy.Status.Replicas)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed getting deployment %s/%s: %w", ns, deploymentName, err)
+	}
```

Comment:
still do we require this one?

Reply:
waitForDeploymentUpdateScale() is only waiting for scale update not for the deployment? I may be wrong.